### DISCUSSION TITLE
profile:hz:99

### DISCUSSION THREAD

**Discussion by yangxingwu** [bpftrace/bpftrace#2428] [Q&A] [answered]
I learn the examples one by one in [tutorial_one_liners](https://github.com/iovisor/bpftrace/blob/master/docs/tutorial_one_liners.md), and got a bit confusing in [lesson-9-profile-on-cpu-kernel-stacks](https://github.com/iovisor/bpftrace/blob/master/docs/tutorial_one_liners.md#lesson-9-profile-on-cpu-kernel-stacks) as the tutorial documenting that **But we don't want 100 exactly, as sampling may occur in lockstep with other timed activities, hence 99.**

```bash
$ bpftrace -e 'profile:hz:99 { @[kstack] = count(); }'
```

so, what does **lockstep** actually mean? would someone explain it with an example?

**Comment by danobi** [comment, marked as answer]
Dictionary says:

> noun: lockstep; noun: lock-step
> a way of marching with each person as close as possible to the one in front.

In this case it probably means that other timed activities may have chosen multiples of 100 as a timer value. Probably b/c multiples of 100 are nice round numbers and humans like that. If the probe chose 100 as a value, we may always run right before or right after such timed activity. This could skew the profiling results, as the profile is no longer randomly sampling activity.

**Reply by yangxingwu** [reply]
thanks @danobi . It makes me more clear about that.
